File: fluttermultiplatform/lib/models/pond.dart
class Pond {
  final String id;
  final String farmId;
  final String name;
  final double area;
  final double depth;
  final String status;
  final String createdAt;

  Pond({required this.id, this.farmId = '', required this.name, this.area = 0, this.depth = 0, this.status = '', this.createdAt = ''});

  factory Pond.fromJson(Map<String, dynamic> json) => Pond(
        id: json['id']?.toString() ?? '',
        farmId: json['farm_id']?.toString() ?? '',
        name: json['name'] ?? '',
        area: (json['area'] ?? 0).toDouble(),
        depth: (json['depth'] ?? 0).toDouble(),
        status: json['status'] ?? '',
        createdAt: json['created_at'] ?? '',
      );
}

File: backend-spring/src/main/java/com/cronos/entity/Pond.java
package com.cronos.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Table(name = "ponds")
@SQLDelete(sql = "UPDATE ponds SET deleted_at = NOW() WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Pond {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "uuid")
    private UUID id;

    @JsonProperty("farm_id")
    @Column(name = "farm_id", nullable = false, columnDefinition = "uuid")
    private UUID farmId;

    @Column(nullable = false, length = 255)
    private String name;

    @Column(precision = 10, scale = 2)
    private BigDecimal area;

    @Column(precision = 5, scale = 2)
    private BigDecimal depth;

    @Column(length = 50)
    @Builder.Default
    private String status = "active";

    @JsonProperty("created_at")
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @JsonProperty("updated_at")
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @JsonIgnore
    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "farm_id", insertable = false, updatable = false)
    @JsonIgnore
    private Farm farm;

    @OneToMany(mappedBy = "pond", fetch = FetchType.LAZY)
    @JsonIgnore
    private List<CultivationCycle> cultivationCycles;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

